### PR TITLE
feat: setup modular monolith with packs-rails and multiple database

### PR DESCRIPTION
**Pull request by ghun131**
## Context

Setting up modular monolith with `packs-rails` and add multiple databases

## How to run
After cloning this repo to your local. Run `./scripts/start.dev.sh`. The migration will probably fail.
- Set up database
```
bin/rails db:setup
```

- Migrate primary database (lago db)
```
bin/rails db:migrate:primary
```

- Migrate publisher_portal db
```
bin/rails db:migrate:publisher_portal
```

- Then rerun `./scripts/start.dev.sh`

- Go to `http://localhost:3000/health` and see this screen. GetLago API is working
image

- Then go to `http://localhost:3000/entitlement/policy` to see this screen. Entitlement module is working
image

- Finally go to `rails console` and paste the code below to mock local data
```ruby
SubscriptionCharge.create(plan_title: 'Title', subscription_instance_id: 1, is_finalized: false)
```
then return to this path `http://localhost:3000/publisher_portal/index` in the browser. This screen should appear
image

### CODE REVIEW

**Review by tnlong311 — CHANGES_REQUESTED** [review]
Please add `# frozen_string_literal: true` at the beginning of the .rb files for styling convention

**Comment by ghun131** [comment]
When you test cases failed at local. You can do these 2 things
1. Add these params to your `.env` file
```bash
LAGO_PDF_URL="https://pdf.lago.dev"
LAGO_API_URL="https://api.lago.dev"
LAGO_FROM_EMAIL="noreply@getlago.com"
```
2. Check you openssl version against ruby openssl version.
- Command to check your openssl version `openssl version`
- Command to check ruby openssl version `rvm reinstall 3.3.0 --with-openssl-dir=/path/to/your/openssl`

File: Gemfile [hunk 1]
```diff
@@ -124,3 +124,6 @@ group :development do
 
   gem 'ruby-lsp-rails', require: false
 end
+
+gem "packs-rails"
+gem "packwerk", "~> 3.2"
```
Comment (TronTram) [inline Gemfile:129]: why do you include packwerk gem in Gemfile while packs-rails already integrate with this gem?

**Review by TronTram — COMMENTED** [review]
a @ghun131 could you please add README for each pack.

File: Gemfile (hunk 1 above)
Comment (ghun131) [reply Gemfile:129]: I don't think `packwerk` already integrate `pack-rails`

**Review by TronTram — APPROVED** [review]

**Pull request merged**